File: backend/src/main/java/com/horain/repository/ProjectRepository.java
package com.horain.repository;

import com.horain.model.Project;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

/**
 * JPA repository for projects.
 */
public interface ProjectRepository extends JpaRepository<Project, UUID> {

    /**
     * Fuzzy search by project name (case-insensitive contains).
     */
    List<Project> findByNameContainingIgnoreCase(String name);
}

File: backend/src/main/kotlin/com/horain/repository/ProjectRepository.kt
package com.horain.repository

import com.horain.model.Project
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

/**
 * JPA repository for projects.
 */
interface ProjectRepository : JpaRepository<Project, UUID> {

    /**
     * Fuzzy search by project name (case-insensitive contains).
     */
    fun findByNameContainingIgnoreCase(name: String): List<Project>
}
